### PR TITLE
Tab select fix

### PR DESCRIPTION
**Pull request by aliabid94**
Fixes issue with tab.select listener not working when it's visiblity is toggled:

Fixes: #7449 Fixes: #7189

### CODE REVIEW

File: js/tabs/shared/Tabs.svelte [hunk 1]
```diff
@@ -31,11 +31,13 @@
 
 	setContext(TABS, {
 		register_tab: (tab: Tab) => {
+			let index: number;
 			let existingTab = tabs.find((t) => t.id === tab.id);
 			if (existingTab) {
 				// update existing tab with newer values
 				let i = tabs.findIndex((t) => t.id === tab.id);
 				tabs[i] = { ...tabs[i], ...tab };
+				index = i;
```
Comment (akx) [inline js/tabs/shared/Tabs.svelte:40]:
```suggestion
				index = tabs.findIndex((t) => t.id === tab.id);
				tabs[index] = { ...tabs[index], ...tab };
```

**Comment by abidlabs** [comment]
Thanks @aliabid94! We've increased what's possible with `gr.Tabs` recently, so it'd be great to add an E2E test for tabs to catch any other regressions that might pop up. Would you be able to create one testing these kinds of workflows:

* Ensuring that tabs interactivity can be toggled on and off
* Ensuring that tabs visibility can be toggled on and off
* Ensuring that toggling visibility does not lose the event listeners
* Ensuring that tabs that are not interactive cannot be navigated to

**Comment by aliabid94** [comment]
ok added all the functional tests @abidlabs

**Comment by abidlabs** [comment]
Cool thanks! We should be good to merge then once they pass

**Pull request merged**